src/resolvers/resolverUtils.ts

```diff
@@ -170,3 +183,101 @@ async function getSchedule(year: number): Promise<Schedule | null> {
     }),
   };
 }
+
+async function createSchedule(
+  year: number,
+  term: Term,
+  scheduleData: ScheduleAlgorithm
+) {
+  const schedule = await initiateSchedule(year);
+
+  scheduleData.summerCourses?.forEach(async (course) => {
+    await upsertCourses(course, term, year, schedule);
+  });
+}
+
+async function getCourseCapacities(input: GenerateScheduleInput) {
+  // TODO: can't be hardcoded for plug and play
+  const algorithm2Url = 'https://algorithm-2.herokuapp.com/predict_class_size';
+  if (!input.courses) return null;
+
+  const request: CourseObject[] = input.courses.map((course) => {
+    return {
+      subject: course.subject,
+      code: course.code,
+      seng_ratio: 0.75,
+      semester: input.term,
+      capacity: 0,
+    };
+  });
+
+  const algorithm2Response = await axios.post<CourseObject[]>(
+    `${algorithm2Url}`,
+    request
+  );
+  return algorithm2Response;
+}
+
+async function generateScheduleWithCapacities(
+  falltermCourses: CourseInput[],
+  summertermCourses: CourseInput[],
+  springtermCourses: CourseInput[],
+  users: User[] | null,
+  capacities: CourseObject[]
+) {
+  // TODO: can't be hardcoded for plug and play
+  const baseUrl = 'https://schedulater-algorithm1.herokuapp.com';
+
+  const courseToCourseInput = (term: Term) => (input: CourseInput) => ({
+    subject: input.subject,
+    courseNumber: input.code,
+    numSections: input.section,
+    courseCapacity:
+      capacities.find((course) => {
+        return (
+          course.subject === input.subject &&
+          course.code === input.code &&
+          course.semester === term
+        );
+      })?.capacity ?? 0,
+    // default capacity to 0 if not found
+    courseTitle: 'testing',
+    // TODO: figure out best value here
+    sequenceNumber: 'testing',
```

Comment:
sequence number is an oddly named variable but from how I've seen alg1 use it, it is supposed to be the section name(eg. A01).

For course title, I can see how that might be an issue, but also algorithm 1 really should not need us to submit this. If for some reason it becomes necessary and is not easily accessible, I might be able to store it in the same json file as the stream sequences. 

No changes needed here, just commenting.